### ISSUE TITLE
green on green lidar or height adjustment 

### ISSUE THREAD

**Issue by bassyboi** [geezacoleman/OpenWeedLocator#146]
keeping boom at correct height in gog applications to allow correct distant for boom height and camera field of view...

**Comment by geezacoleman** [comment]
Yes, this is definitely a challenge. My feeling is lidar might be overkill for it? Perhaps a ToF sensor or ultrasonic sensor are the easiest ways to do it. I think commercial booms use ultrasonic sensors spaced out over the length for this.

**Comment by bassyboi** [comment]
ill checkout how the new jd ploughs do it have used one that is called accudepth i will investigate

**Comment by bassyboi** [comment]
TF-luna Lidar Sensor 0.2m-8m maybe something like this +- 6cm accuracy 

i know holley now make a similar sensor for there new fuel tank gauge that is lidar  the new jd ploughs do have them and they work great

**Comment by geezacoleman** [comment]
This is pretty neat and quite cost effective too. How would you control boom height with this? Alternatively if you know boom height at different locations, you can better map detections to actual location, rather than the basic way it works currently.

**Comment by bassyboi** [comment]
I seen something even neater then this with object detection and depth cameras might be expensive!!! 

I’d just set a actuator up or solenoids up and down 50 mm+-  threshold but my still need to think more about this

**Comment by EchelonMirror** [comment]
I've been working on experimentation with a bunch of ToF sensors. Most of my attention has been paid on the VL53 series of sensors. Most of these have decent enough range (1mm to 2000mm), which is ample if you're looking straight down. If you're looking forward at an angle you're going to need to estimate the range - middle school trig finally coming in useful. 

One thing that is an issue is crop stubbles throwing some height noise on some sensors. This can be resolved easily with some sensors as they have built in noise filters which give you some better accuracy. 

Having one sensor in the same camera box and generating an aggregate number for each boom section (as some booms have double or triple rams for an articulated wing), would give you a far better projection of the height off the ground.
